engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/AbstractTableLocationProvider.java

```diff
@@ -211,30 +211,37 @@ public TableLocation getTableLocationIfPresent(@NotNull final TableLocationKey t
         }
     }
 
-    @Override
+    /**
+     * Remove the given table location.
+     *
+     * @apiNote Use with caution: When this is called all subscribers to the specified location will be notified,
+     *          causing them to stop updating. Tables backed by those notifications will end up 'failed'. This location
+     *          provider will continue to update other locations and will no longer provide or request information about
+     *          the deleted location.
+     *
+     * @param locationKey the TableLocation to remove
```

Comment:
This is missing our discussed warnings, and otherwise is a bit off-message for this layer.
```suggestion
     * Remove a {@link TableLocationKey} and its corresponding {@link TableLocation} (if it was created). All
     * subscribers to this TableLocationProvider will be
     * {@link TableLocationProvider.Listener#handleTableLocationKeyRemoved(ImmutableTableLocationKey) notified}. If the
     * TableLocation was created, all of its subscribers will additionally be
     * {@link TableLocation.Listener#handleUpdate() notified} that it no longer exists. This TableLocationProvider will
     * continue to update other locations and will no longer provide or request information about the removed location.
     * 
     * <p>
     * In practice, there are three downstream patterns in use.
     * <ol>
     * <li>Intermediate TableLocationProviders, which will simply pass the removal on</li>
     * <li>{@link io.deephaven.engine.table.impl.SourceTable SourceTables}, which will propagate a failure notification
     * to all downstream listeners, and become {@link Table#isFailed() failed}.</li>
     * <li>{@link io.deephaven.engine.table.impl.SourcePartitionedTable SourcePartitionedTables}, which will notify
     * their downstream consumers of the removed constituent.</li>
     * </ol>
     * 
     * @apiNote <em>Use with caution!</em> Implementations that call this method must provide certain guarantees: Reads
     *          should only succeed against removed locations if they will return complete, correct, consistent data.
     *          Otherwise, they should fail with a meaningful error message.
     * 
     * @param locationKey The {@link TableLocationKey} to remove
```